### PR TITLE
Exception catch implementation

### PR DESCRIPTION
**Pull request by weilixu**
Exception handlings, should work with getattr_ to remove the requirement of None checks

### CODE REVIEW

**Review by yunjoonjung-PNNL — APPROVED** [review]

File: rct229/rule_engine/rule_base.py [hunk 1]
```diff
@@ -113,26 +114,33 @@ def evaluate(self, rmrs, data=None):
...
-                    else:
-                        # Evaluate the actual rule check
-                        result = self.rule_check(context, calc_vals, data)
-                        if isinstance(result, list):
-                            # The result is a list of outcomes
-                            outcome["result"] = result
-                        # Assume result type is bool
-                        elif result:
-                            outcome["result"] = "PASSED"
+                    try:
```
Comment (charliepnnl) [inline rct229/rule_engine/rule_base.py:117]: I think this try should be moved up to include the applicability check.
The except clauses should then be moved down to match.

File: rct229/rule_engine/rule_base.py [hunk 2]
```diff
@@ -113,26 +114,33 @@ def evaluate(self, rmrs, data=None):
...
+                                outcome["result"] = "FAILED"
+                    except MissingKeyException as ke:
+                        outcome["result"] = "UNDETERMINED"
+                        outcome["message"] = str(ke)
+                    except RCTFailureException as fe:
+                        outcome["result"] = "FAILED"
+                        outcome["message"] = str(fe)
                 else:
-                    outcome["result"] = "NA"
+                    outcome["result"] = "UNDETERMINED"
```
Comment (charliepnnl) [inline rct229/rule_engine/rule_base.py:143]: Do we really want to make this UNDETERMINED?  It seems that NA is distinctly different. I suggest leaving is a NA until a further discussion with Supriya about it.

**Review by charliepnnl — APPROVED** [review]
I'm still not convinced of the need for an exception status, but it isn't hearting anything.

File: rct229/rule_engine/rule_base.py (hunk 1 above)
Comment (weilixu) [reply rct229/rule_engine/rule_base.py:117]: > I'm still not convinced of the need for an exception status, but it isn't hearting anything.

I can comment it out for now.

Comment (weilixu) [reply rct229/rule_engine/rule_base.py:117]: Make sense, I will move the try and catch to include applicability check

**Pull request merged**